src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/submission/AbstractAnalysisSubmission.java

```diff
@@ -0,0 +1,308 @@
+package ca.corefacility.bioinformatics.irida.model.workflow.submission;
+
+import ca.corefacility.bioinformatics.irida.model.IridaResourceSupport;
+import ca.corefacility.bioinformatics.irida.model.MutableIridaThing;
+import ca.corefacility.bioinformatics.irida.model.project.ReferenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hibernate.annotations.Type;
+import org.hibernate.envers.Audited;
+import org.hibernate.envers.NotAudited;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.*;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Abstract superclass for analysis submissions.
+ */
+@Entity
+@Table(name = "analysis_submission")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@Audited
+@EntityListeners(AuditingEntityListener.class)
+public abstract class AbstractAnalysisSubmission extends IridaResourceSupport implements MutableIridaThing {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id")
+	protected Long id;
+
+	@NotNull
+	@Size(min = 3)
+	@Column(name = "name")
+	protected String name;
+
+	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.DETACH, optional = false)
+	@JoinColumn(name = "submitter", nullable = false)
+	protected User submitter;
+
+	/**
+	 * Defines the id of an installed workflow in IRIDA for performing this analysis.
+	 */
+	@NotNull
+	@Column(name = "workflow_id")
+	@Type(type = "uuid-char")
+	protected UUID workflowId;
```

Comment:
My favourite option would be adding support for running old pipelines (see #76), but we likely won't get to that any time soon.

Storing `AnalysisType` and having the params all break is an issue where someone will have to re-create the pipeline, but it would break in a weird (and hard to track) way.

If the pipeline it gets updated and the UUID is no longer applicable, it would also break but would be easier to figure out programatically what's going on.  I'm leaning to keep storing the UUID, and adding some way of disabling automated pipelines with an old version + notifying the user.  If we're able someday to run old pipelines then we're still set up for that.

Reply:
I added a task in #240 to remind me to disable old pipelines when they're upgraded.  That would be done in a later merge.

Reply:
I agree. Storing the UUID is the better option. Thanks.